### ISSUE TITLE
Accessibility: Remove coupon link should be a button and have a descriptive label on legacy Cart and Checkout pages

### ISSUE THREAD

**Issue by Manussakis** [woothemes/woocommerce#56872]
### Prerequisites (mark completed items with an [x]):

- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:
The remove coupon link on the classic Cart/Checkout pages should be a button since clicking on the link doesn't change the current URL. Besides that, screen readers only read "Remove" which fails the [3.3.2 Label or Instruction (Level A)](https://www.w3.org/WAI/WCAG22/Understanding/labels-or-instructions) WCAG criterion.

### Component or Page:
Component: Cart Totals
Pages: Legacy Cart/Checkout pages

![Image](https://github.com/user-attachments/assets/0a17a3de-39fd-44e7-b758-579d16e47146)

### Relevant code snippets:
https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/wc-cart-functions.php#L318
```php
$coupon_html          = $discount_amount_html . ' <a href="' . esc_url( add_query_arg( 'remove_coupon', rawurlencode( $coupon->get_code() ), Constants::is_defined( 'WOOCOMMERCE_CHECKOUT' ) ? wc_get_checkout_url() : wc_get_cart_url() ) ) . '" class="woocommerce-remove-coupon" data-coupon="' . esc_attr( $coupon->get_code() ) . '">' . __( '[Remove]', 'woocommerce' ) . '</a>';
```

### Recommended fix:
- Add the `role="button"` attribute to the remove coupon link. Replacing the `<a>` anchor tag with the `<button>` one is not recommended as this change will cause regressions.
- Add a accessible label to the button that better describes its purpose. e.g., `Remove the {name of the coupon} coupon`.
